File: Client/WebClient/src/app/models/dtos/rtu/rtuMonitoringSettingsDto.ts
import { MonitoringMode } from "../../enums/monitoringMode";
import { PortMonitoringMode } from "../../enums/portMonitoringMode";
import { Frequency } from "../../enums/frequency";
import { RtuMaker } from "../../enums/rtuMaker";
import { OtauPortDto } from "../../underlying/otauPortDto";

export class RtuMonitoringSettingsDto {
  rtuTitle: string;
  rtuMaker: RtuMaker;
  monitoringMode: MonitoringMode;

  preciseMeas: Frequency;
  preciseSave: Frequency;
  fastSave: Frequency;

  lines: RtuMonitoringPortDto[];
}

export class RtuMonitoringPortDto {
  port: string;
  traceId: string;
  otauPortDto: OtauPortDto;
  traceTitle: string;
  portMonitoringMode: PortMonitoringMode;
  durationOfPreciseBase: number;
  durationOfFastBase: number;
}

File: Shared/Dto/ForWebApi/Rtu/RtuMonitoringSettingsDto.cs
﻿using System.Collections.Generic;

namespace Iit.Fibertest.Dto
{
    public class RtuMonitoringSettingsDto
    {
        public string RtuTitle;
        public RtuMaker RtuMaker;
        public string OtdrId;
        public string OtauId;
        public MonitoringState MonitoringMode;

        public Frequency PreciseMeas;
        public Frequency PreciseSave;
        public Frequency FastSave;

        public List<RtuMonitoringPortDto> Lines;
    }
}